packages/dict/src/dict.ts

```diff
@@ -0,0 +1,54 @@
+import { Predicate, Guard, Reducer, UnaryFn } from '@proem/function'
+
+export interface Dictionary {
+  [key:string]: any
+}
+
+export const map = <A extends Dictionary, B>(dict: A, mapfn: ([k, v]: [string, any]) => B): B[] => {
```

Comment:
I think that the dictionary value shouldn't be generic, but the values it contains should be. Also, the mapping function should just take 2 arguments instead of a single tuple.  `<A, B>(dict: Dictionary<A>, mapper: (key: string, value: A) => B)`

Also, since we're mapping dictionaries, shouldn't the return value be a `Dictionary<B>`?

Reply:
@katis I'm not entirely sure what a mapping function over a dictionary should return. In theory you could want to modify the keys of the dictionary so the mapping function should return a key, val tuple?